src/rpc/blockchain.cpp

```diff
@@ -795,6 +797,199 @@ static RPCHelpMan pruneblockchain()
+
+            file << strprintf("height,"
+                              "block_hash,"
+                              "timestamp,"
+                              "mediantime,"
+                              "version,"
+                              );
+
+            if (cumulative) {
+                file << strprintf("total_txs,"
+                                  "total_txouts,"
+                                  "total_coinbase_amount,"
+                                  "total_subsidy,"
+                                  "total_amount_generated,"
+                                  "total_unclaimed_rewards,"
+                                  "total_amount_spent,"
+                                  "total_new_outputs_ex_coinbase_amount,"
+                                  "total_unspendable_scripts,"
+                                  "total_unspendable_amount,"
+                                  "total_genesis_block,"
+                                  "total_bip30,"
+                                  "total_chainwork,"
+                                  "muhash,"
+                                  "total_bogosize\n");
+            } else {
+                file << strprintf("txs,"
+                                  "txouts,"
+                                  "coinbase_amount,"
+                                  "subsidy,"
+                                  "amount_generated,"
+                                  "unclaimed_rewards,"
+                                  "amount_spent,"
+                                  "new_outputs_ex_coinbase_amount,"
+                                  "unspendable_scripts,"
+                                  "unspendable_amount,"
+                                  "genesis_block,"
+                                  "bip30,"
+                                  "chainwork,"
+                                  "muhash,"
+                                  "bogosize\n");
+            }
+
+            CCoinsStats stats{CoinStatsHashType::MUHASH};
+            NodeContext& node = EnsureAnyNodeContext(request.context);
+            ChainstateManager& chainman = EnsureChainman(node);
+            CChainState& active_chainstate = chainman.ActiveChainstate();
+            active_chainstate.ForceFlushStateToDisk();
+            CBlockIndex *pindex = active_chainstate.m_chain.Genesis();
+
+            CCoinsView* coins_view;
+            BlockManager* blockman;
+            {
+                LOCK(::cs_main);
+                coins_view = &active_chainstate.CoinsDB();
+                blockman = &active_chainstate.m_blockman;
+            }
+
+            CCoinsStats prev_stats{CoinStatsHashType::MUHASH};
+            uint64_t num_tx_sum = 0;
+
+            for (; pindex; pindex = active_chainstate.m_chain.Next(pindex)) {
+                if (!GetUTXOStats(coins_view, *blockman, stats, node.rpc_interruption_point, pindex))
+                    throw JSONRPCError(RPC_INTERNAL_ERROR, "Unable to read UTXO set");
```

Comment:
nit: Perhaps the message could be more accurate?
```suggestion
                    throw JSONRPCError(RPC_INTERNAL_ERROR, "Failed to generate coins stats");
```

Reply:
Thanks for the quick re-review! I adapted your suggestion by adding the height of the failure as well, which should make is more helpful in case of corruption in the index.